pkg/networkserver/downlink.go

```diff
@@ -2273,3 +2309,83 @@ func (ns *NetworkServer) processDownlinkTask(ctx context.Context, consumerID str
 	}
 	return err
 }
+
+func (ns *NetworkServer) createProcessPendingDownlinkTask(consumerID string) func(context.Context) error {
+	return func(ctx context.Context) error {
+		return ns.processPendingDownlinkTask(ctx, consumerID)
+	}
+}
+
+// processPendingDownlinkTask processes the most recent pending downlink task ready for execution,
+// if such is available or wait until it is before processing it.
+// NOTE: ctx.Done() is not guaranteed to be respected by processPendingDownlinkTask.
+// The processPendingDownlinkTask receives the consumerID that will be used for popping
+// from the pending downlink task queue.
+func (ns *NetworkServer) processPendingDownlinkTask(ctx context.Context, consumerID string) error {
+	var setErr bool
+	err := ns.pendingDownlinkTasks.Pop(
+		ctx,
+		consumerID,
+		func(ctx context.Context, devID *ttnpb.EndDeviceIdentifiers, t time.Time,
+		) (time.Time, error) {
```

Comment:
nit: The formatting is a bit weird.


```suggestion
		func(
		    ctx context.Context, 
		    devID *ttnpb.EndDeviceIdentifiers, 
		    t time.Time,
		) (time.Time, error) {
```

or


```suggestion
		func(ctx context.Context, devID *ttnpb.EndDeviceIdentifiers, t time.Time) (time.Time, error) {
```

Reply:
fixed